src/modules/credentials/services/CredentialService.ts

```diff
@@ -403,12 +429,15 @@ export class CredentialService {
     const { comment } = options
     const credentialRequest = new RequestCredentialMessage({
       comment,
-      requestAttachments: [attachment],
+      requestAttachments: [requestAttachment],
+      attachments: credentialRecord.offerMessage?.attachments,
```

Comment:
This seems wrong. the attachments array can contain any attachment, not just linked attachments. We should only send the linked attachments

Reply:
I am not quite sure what you mean by this. If you mean that `credentialRecord.offerMessage?.attachments` can contain attachments that are not linked, then I think it should be fixed when the attachments are appended to the `offerMessage`. However, why not any attachment?

Reply:
It is not standard practice to include all attachments in all messages. Different use cases require different flows. One of them could be adding an attachment to the offer message only. I would find it weird as a framework user if it is then automatically sent in the credential request. That should be a manual step.

I want to keep the logic of handling linked attachments (which have the process defined) over general attachments (which are general purpose, and don't have the process defined)

Reply:
Okay yeah that sounds good.